src/test/java/org/broadinstitute/hellbender/tools/spark/pipelines/PrintReadsSparkIntegrationTest.java

```diff
@@ -67,6 +74,44 @@ public void testGCSInputsAndOutputsWithSparkNio(final String gcsInput, final Str
         SamAssertionUtils.assertSamsEqual(IOUtils.getPath(outputPath), IOUtils.getPath(gcsInputPath), null);
     }
 
+    @Test(groups = "spark")
+    public void testReadAndWriteCRAMAndReferenceOnHDFS() throws Exception {
+        final GATKPathSpecifier testCram = new GATKPathSpecifier(getTestFile("count_reads.cram").getAbsolutePath());
+        final GATKPathSpecifier testRef = new GATKPathSpecifier(getTestFile("count_reads.fasta").getAbsolutePath());
+        final GATKPathSpecifier testRefDict = new GATKPathSpecifier(getTestFile("count_reads.dict").getAbsolutePath());
+        final GATKPathSpecifier testRefIndex = new GATKPathSpecifier(getTestFile("count_reads.fasta.fai").getAbsolutePath());
+
+        MiniClusterUtils.runOnIsolatedMiniCluster(cluster -> {
+            final org.apache.hadoop.fs.Path workingDirectory = MiniClusterUtils.getWorkingDir(cluster);
+
+            // copy all the test files to HDFS
+            final org.apache.hadoop.fs.Path cramHDFSPath = new org.apache.hadoop.fs.Path(workingDirectory,"count_reads.cram");
+            final org.apache.hadoop.fs.Path refHDFSPath = new org.apache.hadoop.fs.Path(workingDirectory, "count_reads.fasta");
+            final org.apache.hadoop.fs.Path refHDFSDictPath = new org.apache.hadoop.fs.Path(workingDirectory,"count_reads.dict");
+            final org.apache.hadoop.fs.Path refHDFSIndexPath = new org.apache.hadoop.fs.Path(workingDirectory, "count_reads.fasta.fai");
+            cluster.getFileSystem().copyFromLocalFile(new org.apache.hadoop.fs.Path(testCram.getURI()), cramHDFSPath);
+            cluster.getFileSystem().copyFromLocalFile(new org.apache.hadoop.fs.Path(testRef.getURI()), refHDFSPath);
+            cluster.getFileSystem().copyFromLocalFile(new org.apache.hadoop.fs.Path(testRefDict.getURI()), refHDFSDictPath);
+            cluster.getFileSystem().copyFromLocalFile(new org.apache.hadoop.fs.Path(testRefIndex.getURI()), refHDFSIndexPath);
+
+            // run PrintReadsSpark and print the contents of the HDFS cram test file to an output HDFS cram
+            final GATKPathSpecifier outputHDFSPath = new GATKPathSpecifier(workingDirectory + "testCramOnHDFSOut.cram");
+            final ArgumentsBuilder argBuilder = new ArgumentsBuilder();
+            argBuilder.add("input", cramHDFSPath.toUri().toString())
```

Comment:
nitpick, i'm a booster for argBuilder.addInput and .addOutput, but it doesn't matter at all

Reply:
I keep forgetting about those. Easy enough to change though.